app/controllers/userController.js

```diff
@@ -0,0 +1,9 @@
+import UserService from '../services/userService'
+
+class UserController {
+  static async signup(req, res) {
+    return UserService.createUser(req, res)
```

Comment:
I think I get what you mean here. 

_you might need to revisit `async/await` though 😄 here is a  [refresher](https://javascript.info/async-await#error-handling)_ 

am thinking of a `CustomError` class that extends the built-in `Error ` that can accept both a `message` and a `status code`. so this class will be used to through errors with an explicit status code as well as the message. This will enable the global error handler defined to return detailed error messages as well as the correct status code instead of the default 500.

so we could have something like this 
```
// userService 
const user = db.User.findOne({where: { email }})
if(user){
   // throw and error here. it will be handled by the try/catch block in the controller and passed to next
   throw new CustomError(409, 'User with given email already exists')
}

// userController
static async createUser(req, res, next){
  try{
  await UserService.createUser(req.body)
  return res.status(201).json({message: 'signup success'})
}catch(error)
// in the case the UserService.createUser method threw an error or something went wrong then it will be caught here .
next(error)
}
```